public/dorequest.php

```diff
@@ -333,8 +333,10 @@ function DoRequestError(string $error, ?int $status = 200, ?string $code = null)
              */
             $response = array_merge($response, unlockAchievement($user, $achIDToAward, $hardcore));
 
-            dispatch(new UnlockPlayerAchievementJob($user->id, $achIDToAward, $hardcore))
-                ->onQueue('player-achievements');
+            if ($foundAchievement->is_published) {
```

Comment:
Would it make sense to check `$response['success']`, which would also prevent firing the job if the player had previously unlocked the achievement?

Reply:
Changed the conditional to simply `$response['Success']` in latest.